api/unstable/npn.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <s2n.h>
+
+/**
+ * @file npn.h
+ *
+ * The Next Protocol Negotiation Extension, or NPN, was an early RFC proposal to
+ * negotiate an application protocol. This proposal was never standardized, and
+ * it was eventually replaced with the ALPN Extension. Unfortunately, NPN
+ * was implemented in several TLS libraries. Now, clients and servers using
+ * older TLS implementations may require this extension in order to connect.
+ *
+ * s2n-tls supports NPN to make it easier for users whose peers require this
+ * extension, but s2n-tls does NOT recommend its use.
+ */
+
+/**
+ * Turns on support for the NPN extension.
+ * 
+ * This will allow an s2n-tls client to send the NPN extension and an s2n-tls
+ * server to respond to receiving the NPN extension. However, if their peer
+ * also indicates support for the ALPN extension, s2n-tls will prefer that.
+ * 
+ * Use s2n_config_set_protocol_preferences() to set up a list of supported protocols.
```

Comment:
Might want to mention `s2n_config_append_protocol_preference`. I think that's our preferred / suggested way to set the protocol preferences.